src/Recurrence.ts

```diff
@@ -212,4 +207,145 @@ export class Recurrence {
+     */
+    private static nextAfterMonths(
+        after: Moment,
+        next: Moment,
+        rrule: RRule,
+        skippingMonths: string | undefined,
+    ): Moment {
+        // Parse `skippingMonths`, if it exists.
+        let parsedSkippingMonths: Number = 1;
+        if (skippingMonths !== undefined) {
+            parsedSkippingMonths = Number.parseInt(skippingMonths.trim(), 10);
+        }
+
+        // While we skip the wrong number of months, move `after` one day into the past.
+        while (Recurrence.isSkippingTooManyMonths(after, next, parsedSkippingMonths)) {
+            // The next line alters `after` to be one day earlier.
+            // Then returns `next` based on that.
+            next = Recurrence.fromOneDayEarlier(after, rrule);
+        }
+
+        return next;
+    }
+
+    /**
+     * isSkippingTooManyMonths returns true if `next` is more than `skippingMonths` months after `after`.
+     */
+    private static isSkippingTooManyMonths(after: Moment, next: Moment, skippingMonths: Number): boolean {
+        let diff = next.month() - after.month();
+        if (diff < 0) {
+            diff = diff + 12;
+        }
+
+        return diff > skippingMonths;
+    }
+
+    /**
+     * nextAfterYears calculates the next date after `skippingYears` years.
+     *
+     * `skippingYears` defaults to `1` if undefined.
+     */
+    private static nextAfterYears(
+        after: Moment,
+        next: Moment,
+        rrule: RRule,
+        skippingYears: string | undefined,
+    ): Moment {
+        // Parse `skippingYears`, if it exists.
+        let parsedSkippingYears: Number = 1;
+        if (skippingYears !== undefined) {
+            parsedSkippingYears = Number.parseInt(skippingYears.trim(), 10);
+        }
+
+        // While we skip the wrong number of years, move `after` one day into the past.
+        while (Recurrence.isSkippingTooManyYears(after, next, parsedSkippingYears)) {
+            // The next line alters `after` to be one day earlier.
+            // Then returns `next` based on that.
+            next = Recurrence.fromOneDayEarlier(after, rrule);
+        }
+
+        return next;
+    }
+
+    /**
+     * isSkippingTooManyYears returns true if `next` is more than `skippingYears` years after `after`.
+     */
+    private static isSkippingTooManyYears(after: Moment, next: Moment, skippingYears: Number): boolean {
+        const diff = next.year() - after.year();
+
+        return diff > skippingYears;
+    }
+
+    /**
+     * fromOneDayEarlier returns the next occurrence after moving `after` one day into the past.
+     *
+     * WARNING: This method manipulates the given instance of `after`.
+     */
+    private static fromOneDayEarlier(after: Moment, rrule: RRule): Moment {
+        after.subtract(1, 'days').endOf('day');
+
+        const options = rrule.origOptions;
+        options.dtstart = after.startOf('day').toDate();
+        rrule = new RRule(options);
```

Comment:
Is this also modifying the supplied `rule`?

Reply:
I don't know, to be honest. But I believe so.